### PR TITLE
feat: Unread activities information Display number of unread activities for a space from the left menu - MEED-700- Meeds-io/MIPs#17

### PR DESCRIPTION
**Pull request by SaraBoutej**
This Task allows to add a badge of the number of unread items per application to the recent viewed spaces from the left navigation menu. For now this information will be retrieve only when the value "unread" is added to "&expand=...,unread" Path parameter.

### CODE REVIEW

**Review by AzmiTouil — APPROVED** [review]

File: component/api/src/main/java/org/exoplatform/social/metadata/MetadataService.java [hunk 1]
```diff
@@ -298,6 +298,17 @@ List<MetadataItem> getMetadataItemsByMetadataNameAndTypeAndObjectAndMetadataItem
...
+   * Count the size of Metadata items to a given {@link Metadata} type,
+   * {@link MetadataItem} creatorId
+   * by a given {@link Space} identifier
+   *
+   * @param metadataTypeName {@link Metadata} type
+   * @param creatorId {@link MetadataItem} creatorId
+   * @param spaceId {@link Space} technical identifier
+   * @return map of Metadata items grouped by objectId {@link MetadataObject}
+   */
+  Map<String, Long> countMetadataItemsByMetadataTypeAndAudienceId (String metadataTypeName, long creatorId, long spaceId);
```
Comment (boubaker) [inline component/api/src/main/java/org/exoplatform/social/metadata/MetadataService.java]:
```suggestion
  Map<String, Long> countMetadataItemsByMetadataTypeAndAudienceId(String metadataTypeName, long creatorId, long spaceId);
```

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/MetadataItemDAO.java [hunk 2]
```diff
@@ -55,6 +55,8 @@ public class MetadataItemDAO extends GenericDAOJPAImpl<MetadataItemEntity, Long>
   
   private static final String SPACE_ID             = "spaceId";
 
+  private static final String UNREAD_METADATA_TYPE = METADATA_TYPE_PARAM;
```
Comment (boubaker) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/MetadataItemDAO.java]: why repeating the constant, you can reuse it instead IMO.

File: component/service/src/main/java/org/exoplatform/social/rest/api/SpaceRestResources.java [hunk 3]
```diff
@@ -62,6 +63,7 @@ public abstract Response getSpaces(@Context UriInfo uriInfo,
                                      @QueryParam("order") String order,
                                      @QueryParam("returnSize") boolean returnSize,
                                      @QueryParam("favorites") boolean favorites,
+                                     @QueryParam("unread") boolean unread,
```
Comment (boubaker) [inline component/service/src/main/java/org/exoplatform/social/rest/api/SpaceRestResources.java]:
```suggestion
```
Please avoid adding a dedicated parameter to retrieve or the unread information, we can use expand parameter.

File: component/api/src/main/java/org/exoplatform/social/metadata/MetadataService.java (hunk 1 above)
Comment (SaraBoutej) [reply component/api/src/main/java/org/exoplatform/social/metadata/MetadataService.java]: Done!

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/MetadataItemDAO.java (hunk 2 above)
Comment (SaraBoutej) [reply component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/MetadataItemDAO.java]: Done!

File: component/service/src/main/java/org/exoplatform/social/rest/api/SpaceRestResources.java (hunk 3 above)
Comment (SaraBoutej) [reply component/service/src/main/java/org/exoplatform/social/rest/api/SpaceRestResources.java]: Done!

File: component/service/src/main/java/org/exoplatform/social/rest/impl/space/SpaceRestResourcesV1.java [hunk 4]
```diff
@@ -170,9 +168,10 @@ public Response getSpaces(@Context UriInfo uriInfo,
           orderBy = Sorting.OrderBy.valueOf(order.toUpperCase());
         }
         spaceFilter.setSorting(new Sorting(sortBy, orderBy));
-    }
-    spaceFilter.setIsFavorite(favorites);
-
+      }
+      if (StringUtils.isNotBlank(expand)) {
+        spaceFilter.setIsFavorite(Arrays.asList(StringUtils.split(expand, ",")).contains(RestProperties.FAVORITE));
+      }
```
Comment (boubaker) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/space/SpaceRestResourcesV1.java]:
```suggestion
    }
    if (StringUtils.isNotBlank(expand)) {
      spaceFilter.setIsFavorite(Arrays.asList(StringUtils.split(expand, ",")).contains(RestProperties.FAVORITE));
    }
```

**Review by boubaker — APPROVED** [review]

**Pull request merged**